fix: @solana/web3.js to 1.98.2 #7221

limichange opened the pull request:
https://security.snyk.io/vuln/SNYK-JS-BIGINTBUFFER-3364597

- Updated the version of @solana/web3.js to 1.98.2 in both package.json and yarn.lock for improved functionality and compatibility.
- Added new dependencies related to @solana/codecs-core and @solana/errors to support the updated version.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Chores**
  - Updated the version of `@solana/web3.js` dependency to `1.98.2` throughout the project to ensure consistency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
chore: update @solana/web3.js to version 1.98.2 in package.json and yarn.lock

- Updated the version of @solana/web3.js to 1.98.2 in both package.json and yarn.lock for improved functionality and compatibility.
- Added new dependencies related to @solana/codecs-core and @solana/errors to support the updated version.

originalix commented on packages/core/package.json:
```diff
@@ -30,7 +30,7 @@
     "@onekeyfe/kaspa-wasm": "^1.0.2",
     "@solana/spl-token": "^0.4.8",
-    "@solana/web3.js": "^1.87.7",
+    "@solana/web3.js": "^1.98.2",
```
```suggestion
    "@solana/web3.js": "1.98.2",
```
这里也固定版本吧，这项目之前出过事

limichange committed:
chore: standardize @solana/web3.js versioning in package.json and yarn.lock

- Changed the version specification of @solana/web3.js from caret (^) to exact version (1.98.2) in both package.json and yarn.lock for consistency and to prevent unintended upgrades.

limichange committed:
Merge branch 'x' into fix-7221

limichange committed:
Merge branch 'x' into fix-7221

limichange committed:
Merge branch 'x' into fix-7221
